flink-end-to-end-tests/flink-plugins-test/src/main/java/org/apache/flink/fs/dummy/DummyFs.java

```diff
@@ -0,0 +1,166 @@
+
+class DummyFs extends FileSystem {
+
+	static final URI FS_URI = URI.create("dummy:///");
+
+	private static final String HOSTNAME = "localhost";
+
+	private final URI workingDir;
+
+	private final URI homeDir;
+
+	private final Map<String, byte[]> contents;
+
+	DummyFs(Map<String, String> contents) {
+		this.workingDir = new File(System.getProperty("user.dir")).toURI();
+		this.homeDir = new File(System.getProperty("user.home")).toURI();
+		this.contents = convertToByteArrayMap(contents);
+	}
+
+	// ------------------------------------------------------------------------
+
+	@Override
+	public URI getUri() {
+		return FS_URI;
+	}
+
+	@Override
+	public Path getWorkingDirectory() {
+		return new Path(workingDir);
+	}
+
+	@Override
+	public Path getHomeDirectory() {
+		return new Path(homeDir);
+	}
+
+	@Override
+	public boolean exists(Path f) throws IOException {
+		return getDataByPath(f) != null;
+	}
+
+	@Override
+	public FileStatus[] listStatus(final Path f) throws IOException {
+		byte[] data = getDataByPath(f);
+		if (data == null) {
+			return null;
+		}
+		return new FileStatus[] { new DummyFsFileStatus(f, data.length) };
+	}
+
+	@Override
+	public BlockLocation[] getFileBlockLocations(FileStatus file, long start, long len) throws IOException {
+		return new BlockLocation[] {
+			new LocalBlockLocation(HOSTNAME, file.getLen())
+		};
+	}
+
+	@Override
+	public FileStatus getFileStatus(Path f) throws IOException {
+		byte[] data = getDataByPath(f);
+		if (data == null) {
+			throw new FileNotFoundException("File " + f + " does not exist or the user running "
+				+ "Flink ('" + System.getProperty("user.name") + "') has insufficient permissions to access it.");
+		}
+		return new DummyFsFileStatus(f, data.length);
+	}
+
+	@Override
+	public FSDataInputStream open(final Path f, final int bufferSize) throws IOException {
+		return open(f);
+	}
+
+	@Override
+	public FSDataInputStream open(final Path f) throws IOException {
+		return DummyFsInputStream.create(getDataByPath(f));
+	}
+
+	@Override
+	public boolean delete(final Path path, final boolean recursive) throws IOException {
+		checkNotNull(path, "path is null");
+		throw new UnsupportedOperationException("Test FS doesn't support delete operation");
+	}
+
+	@Override
+	public boolean mkdirs(final Path path) throws IOException {
+		checkNotNull(path, "path is null");
+		throw new UnsupportedOperationException("Test FS doesn't support mkdirs operation");
+	}
+
+	@Override
+	public FSDataOutputStream create(final Path path, final WriteMode overwrite) throws IOException {
+		checkNotNull(path, "path");
+		throw new UnsupportedOperationException("Test FS doesn't support create operation");
+	}
+
+	@Override
+	public boolean rename(final Path src, final Path dst) throws IOException {
+		throw new UnsupportedOperationException("Test FS doesn't support rename operation");
+	}
+
+	@Override
+	public boolean isDistributedFS() {
+		return false;
```

Comment:
shouldn't this be `return true`? All of the processes will see the same dummy files, right?

Reply:
I haven't dug into semantics of this flag and how it is interpreted internally. The original implementation is heavily based on `LocalFileSystem`. But I have changed it to `true`, the batch wordcount test still pass.

Reply:
I guess it should be fine either way, but to me it looks like `true` would be more accurate and potentially less confusing for future developers